addons/auth_signup/models/res_users.py

```diff
@@ -213,10 +221,13 @@ def _action_reset_password(self, signup_type="reset"):
                 raise UserError(_("Cannot send email: user %s has no email address.", user.name))
             email_values['email_to'] = user.email
             with contextlib.closing(self.env.cr.savepoint()):
+                is_portal = user._is_portal()
```

Comment:
Use the reverse "_is_internal", that way we correctly filter out portal / public / no groups / .. whatever could happen.